### PR TITLE
[CR] Add new armour-grade bone material from skeleton enemies

### PR DESCRIPTION
**Pull request by Photoloss**
#### Summary
```SUMMARY: Content "Adds new material for bone armour"```

#### Purpose of change
More elaborate "in-universe realistic" alternative to #28064. Rather than removing bone armor entirely the material used must now be sourced from the plating of skeleton-type enemies which demonstrate its effectiveness.

#### Describe the solution
- Add new material "Petrified Bone" (name pending), implied to be unnaturally resilient and thus armour-viable due to Blob mutation. 
- Shifted 1 point of material resist from "bash" to "cut" to better reflect its performance on enemies. Increased density by 1 point.
- Butchery yield for skeleton-type enemies adjusted to yield new armour-grade bone.
- Both personal and vehicle bone armor now uses this new material in its creation with no further changes to their stats.
- Tools, foods and other bone crafts not updated so far.
- Regular bone material remains unchanged as all combat-relevant inclusions should have been eliminated. Future designers of tribal fashion are kindly asked to keep non-ceremonial uses rare by choosing sensible material thicknesses.
- Added survival skill requirement equal to tailoring primary in armour crafting as you are handling exotic creature remains.

#### Describe alternatives you've considered
- Material type for skeleton enemies themselves not updated as I am unsure where else this value might be used (e.g. break soundpacks or so)
- Zombie Hulks and possibly other "super-strong" variants like Brutes and Shocker Brutes (EDIT: zombie wrestler and nightstalker also get the "SMASH" attack) also require an enhanced skeletal structure to survive the reaction force from their punches. I'd like to add hardened bone to their butchery yield as well if the base change is accepted. Please mention any other enemies with a "megaton punch" attack or similar needs for stronger endoskeletons.
- If anyone knows a JSON-only fix to prevent "invalid material" crashes please mention it, I found the hardcoded "basic repair" checks and that fixed it on my end but I might have missed a more flexible overhead.

#### Additional context
I kindly request a canonical lore explanation as to *how* skeleton enemies are supposed to achieve their extremely resilient natural armour. My current assumption being that the bone is compacted into a denser, sturdier but more brittle configuration closer to stone. Hence it would chip and break easily when you attempt to make a needle out of it and without marrow etc. its nutritional value is essentially equal to Blob-tainted limestone.

### CODE REVIEW

**Comment by KITbutler** [comment]
Okay I see the astyle of "item_factory.cpp" failing in gorgon, but when I locally run "make clean"-> "make astyle" it does not change the appearance of the file. No errors and deliberate screw-ups like deleting spaces *are* reverted so could someone please explain what is going on here?

**Comment by Photoloss** [comment]
Okay I see the astyle of "item_factory.cpp" failing in gorgon, but when I locally run "make clean"-> "make astyle" it does not change the appearance of the file. No errors and deliberate screw-ups like deleting spaces _are_ reverted so could someone please explain what is going on here?

**Comment by kevingranade** [comment]
Not a fan in general.
1. Why is the coverage so high?
2. It's not light any more, make it heavy.
3. We generally avoid harvesting things from zombies under the rationales:
3a. They're held together by blob and will stop working immediately upon death.
3b. From a game balance POV the world is flooded with them, so being able to craft with their products means you can easily amass huge amounts of wealth.

File: src/item_factory.cpp [hunk 1]
```diff
@@ -376,7 +376,8 @@ void Item_factory::finalize_post( itype &obj )
     if( !obj.gun && !obj.count_by_charges() && std::any_of( obj.materials.begin(), obj.materials.end(),
     []( const material_id & m ) {
     return m == material_id( "wood" ) || m == material_id( "paper" ) ||
-               m == material_id( "bone" ) || m == material_id( "chitin" ) || m == material_id( "acidchitin" );
+               m == material_id( "bone" ) || m == material_id( "bone_hard" ) ||
```
Comment (AMurkin) [inline src/item_factory.cpp]:
```suggestion
               m == material_id( "bone" ) || m == material_id( "bone_hard" ) ||
```

**Comment by Photoloss** [comment]
TL;DR: since you **DO** seem to have balance concerns regarding the present state of the armour could you please voice those openly instead of arguing *solely* based on real-life realism for its removal?

>     1. Why is the coverage so high?

Because I did not change any stats on the armour itself. Your own assessment in PR #28064 did not mention any gameplay or balance concerns so I assumed those aspects to be acceptable by default.

In-universe and reviewing the stats the helmet having 100% coverage definitely is too high. The 95% on the other pieces are in line with chainmail and plate armour which honestly sounds reasonable to me since this gear would be tailor-made to fit the player and cover as much as possible. Crafting times might warrant an increase though as RL armour crafting was extremely labour-intensive.

>     2. It's not light any more, make it heavy.

Mechanically it's still "half" leather to my understanding and I only increased the density of the bone by ~15% awaiting a canonical description of skeleton armour. Apparently the RL [density of bone](https://www.aqua-calc.com/page/density-table/substance/bone) is around 2g/cm³ but since the boots alone take up 4l of volume that won't help without additional information. How thick is one unit of "material thickness" in physical units, or is there some other guideline for armour?

For reference: density of bone armor suit = 0.21g/cm³ or 10% of bone material, density of plate armor = 0.63g/cm³ or **8% of pure RL steel!** Then again in-game steel scrap varies between 0.2g/cm³ (scrap) and 4.0g/cm³ (lump) (wikipedia gives 8g/cm³) so at this point I have no idea what the justification for any weight or density is anymore.

>        3a. They're held together by blob and will stop working immediately upon death.

This clearly is not the case. "Metabolic" processes might cease and pulping-level damage prevents regeneration but tainted meat retains its mutagenic properties and tainted bones from **any** zed can be crafted into small tools. Outright Nether mobs are the ones lacking ontological inertia and this armour is not implied or programmed to intelligently adapt, regenerate from damage or otherwise require an active Blob component to function.

>        3b. From a game balance POV the world is flooded with them, so being able to craft with their products means you can easily amass huge amounts of wealth.

Err, what?! **ANY** zombie yields rags, probably leather+plastic and a chance at important tools, batteries, scrap metal and even gun ammo. This is only *one armour set* available from *specific* enemies, ones which are [harder to kill](https://github.com/CleverRaven/Cataclysm-DDA/blob/master/data/json/monsters.json#L846) than the [infinitely respawning ants](https://github.com/CleverRaven/Cataclysm-DDA/blob/master/data/json/monsters/insect_spider.json#L735).

File: src/item_factory.cpp (hunk 1 above)
Comment (Photoloss) [reply src/item_factory.cpp]: Thanks!

**Comment by paulenka-aleh** [comment]
In addition to what has been said I hardly can imagine a survivor being able to figure out it's good for armor, being able to figure out how to harvest it and how to craft from it. Given its properties it doesn't sound easy.

**Comment by Photoloss** [comment]
> In addition to what has been said I hardly can imagine a survivor being able to figure out it's good for armor, being able to figure out how to harvest it and how to craft from it. Given its properties it doesn't sound easy.

You *have* to kill the things using it **as armour** to gather the material. Harvesting works the same as butchering any other large creature, if we can add a bone saw requirement to the butchery that would be fine but beyond the scope of this PR (good luck taking apart a moose or giant bug if you're struggling with a bone-plated dog)

The crafting part I *can* address here and the original recipes do seem excessively simple. Besides the already mentioned time increase I imagine you'd need a saw and a drill (do we have anything to mimic a file?) plus **a lot** of thread to connect up individual plates. Anything else you'd definitely need to create a construction similar to [brigandine](https://en.wikipedia.org/wiki/File:Brigandine,_Italian,_c1470,_Royal_Armoury,_Leeds_(internal_view).JPG) or [lamellar](https://en.wikipedia.org/wiki/Lamellar_armour) armour?

**Comment by Xhuis** [comment]
I think what a survivor knows is passed over for gameplay reasons, and I think it's acceptable in this case too. Keep in mind, this is the same survivor who knows how to cook entire books' worth of food by default, and can transform acidic chitin from mutated ants into wearable armor.

**Comment by paulenka-aleh** [comment]
> acidic chitin
To me it seems questionable as well. Should survivor be able to?.. The only justification I see is balance: now you have a reward for going into acid anthill. I'd rather see merged acid ant solders into normal anthills to make them more a dangerous place and less a source of free meat, chitin and egg powder with acid anthill and acidic chitin removed from the game. Doesn't bother me a lot though.

But it does when it is used as a justification to add in game something like this.

**Comment by Photoloss** [comment]
> But it does when it is used as a justification to add in game something like this.

First of all I am not **adding** this I am reworking something which was already mainlined before I even started playing. Secondly the principle behind lamellar/brigandine construction is not particularly complex, the [article](http://siberiantimes.com/science/casestudy/features/warriors-3900-year-old-suit-of-bone-armour-unearthed-in-omsk/) Kevin himself linked on bone armour describes a find which is **4000** years old! The current skill, tool and time requirements are bonkers low however and the same probably goes for both chitin sets. I don't know how large armour-grade chunks of chitin actually react to various tool applications though so if you find some sources please open an issue and post them.

**Comment by paulenka-aleh** [comment]
> reworking something which was already mainlined

So after your and Kevin's PRs get merged my innawoods survivor won't be able to craft bone armor after butchering some animal and make something weak and probably not very useful as he probably should be given his high survival, fabrication and tailoring skills. On the other had he can visit some dangerous blob-infeasted location, butcher some Zs and get matetial that doesn't exist IRL and not even well-explained by lore and than craft from it more or less useful armor that resembles current bone armor?

Well, I'd call it "adding an alternative to something that is entirely removed" rather than rework.

**Comment by paulenka-aleh** [comment]
> reworking something which was already mainlined

So after your and Kevin's PRs get merged my innawoods survivor won't be able to craft bone armor after butchering some animal and make something weak and probably not very useful as he probably should be given his high survival, fabrication and tailoring skills. On the other had he can visit some dangerous blob-infeasted location, butcher some Zs and get matetial that doesn't exist IRL and not even well-explained by lore and than craft from it more or less useful armor that resembles current bone armor?

Well, I'd call it "adding an alternative to something that is entirely removed" rather than rework.

**Comment by Photoloss** [comment]
> So after your and Kevin's PRs get merged my innawoods survivor won't be able to craft bone armor after butchering some animal and make something weak and probably not very useful as he probably should be given his high survival, fabrication and tailoring skills. 

Good point actually if leather armor cannot fill that niche. What would be the skill level and a comparable existing armour you envision for "early-game" innawoods? The bone suit is 6 tailoring right now which is high enough to craft survivor armor from heli crashes.

> On the other had he can visit some dangerous blob-infeasted location, butcher some Zs and get matetial that doesn't exist IRL and not even well-explained by lore and than craft from it more or less useful armor that resembles current bone armor?

That second part is not my fault, I asked for clarification. Something with these vague specs clearly **does** exist in-game however so the IRL argument does not fly unless the skeleton enemies are also reworked or removed.

**Comment by kevingranade** [comment]
You know what, I'm not even going to entertain this right now.  It's clearly a knee-jerk attempt to work around the rationale for removing bone armor with no real thought put into it, and as you keep deflecting instead of addressing any concerns this is more trouble than it's worth to deal with.

I voiced my concerns in case you had a cogent argument for its inclusion, but nothing has appeared.

**Comment by Xhuis** [comment]
Man, what the heck is your problem? Can't someone want something to keep existing in the game? I'd like to believe he could have addressed those concerns, but I've seen this cycle repeated over and over, and it ends up being conceded aspect after conceded aspect until either the author gives up or the change no longer even remotely resembles what it originally was.

You've gone from suggesting an alternative in your PR to remove bone armor (making it cosmetic) to outright saying it doesn't belong in the game. If you were so against having it to begin with, why not just remove it in a direct commit instead of leading people along that it could have a chance of existing?

**Comment by Photoloss** [comment]
Kevin, Quit. Your. Bullshit.

> It's clearly a knee-jerk attempt to work around the rationale for removing bone armor

It is indeed, seeing as *your own* PR for its removal lists no well-argued precedent and the one argument you give is easily refuted at the qualitative level. *Neither* of us has come up with any numbers on *how* strong bone should be.

> as you keep deflecting instead of addressing any concerns

**WHAT?!** I *asked for clarification* both on lore:

> I kindly request a canonical lore explanation as to _how_ skeleton enemies are supposed to achieve their extremely resilient natural armour.

and on quantitative game mechanics:

> How thick is one unit of "material thickness" in physical units, or is there some other guideline for armour?

And you denied the former then closed the PR without answering the latter either. I conceded the crafting difficulty:

> being able to figure out how to harvest it and how to craft from it. Given its properties it doesn't sound easy.
> The crafting part I _can_ address here and the original recipes do seem excessively simple. Besides the already mentioned time increase I imagine you'd need a saw and a drill (do we have anything to mimic a file?) plus **a lot** of thread to connect up individual plates. Anything else you'd definitely need to create a construction similar to brigandine or lamellar armour?

I addressed your own post to the extent I could without further input. You did not give any stats or other quantitative input though so immediate well-balanced changes were not possible without reading your mind.

I **HAD** another commit in the works nerfing the helmet coverage and increasing weight by a factor of 1.5 to start with, but that is when I started comparing to in-game leather armor and noticed the two were pretty much on par already, having trusted *your* assessment of bone having undeservedly better stats until then. So that obviously froze everything because I would have to go against **YOUR listed alternative** from the original PR. Then you closed this without addressing any points I'd made (not *answering* them with a full research paper but at least pointing out any obvious flaws, like the "living vs dead bone" thing) but accuse **ME** of not having a "cogent argument".

**Comment by thethunderhawk** [comment]
>...for committing the crime of not being realistic in a game with cybernetic augments and mutation goo made from sewer water.

I just wanna point out for anyone reading that this kind of argument has never flown around here. The goal is for the game to be as realistic as possible *despite the sci-fi setting*. The very point of the nerfs and removals are to make things like that feel more in line with the expected realism level of the game. In general, trying to sell one unrealistic idea by pointing to another does not fly.

**Comment by dyingsoundsfun** [comment]
Perhaps an atom suddenly splits due to massive emotional trauma deep within
Wolverine devastating past, causing his adamantium skeleton to be turned
inside out like the cone of an Explosive Formed Projectile used against
dumb stoner dropout GIs in Iraq like woah. This then turnz into a sweet azz
beetle like exoskeleton
